metricbeat/scripts/mage/modules_collector.go

```diff
@@ -0,0 +1,100 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package mage
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// CollectModules collects module configs to modules.d
+func CollectModules() error {
+	header := `# Module: %[1]s
+# Docs: https://www.elastic.co/guide/en/beats/%[2]s/%[3]s/%[2]s-module-%[1]s.html
+
+`
+	r, err := regexp.Compile(`.+\.reference\.yml`)
+	if err != nil {
+		return err
+	}
+
+	beatName := os.Getenv("BEAT_NAME")
+	docsBranch := os.Getenv("DOCS_BRANCH")
+
+	path, err := filepath.Abs("module")
+	if err != nil {
+		return err
+	}
+
+	modules, err := ioutil.ReadDir("module")
+	if err != nil {
+		return err
+	}
+
+	if err = os.Mkdir("modules.d", os.ModeDir|os.ModePerm); err != nil {
```

Comment:
Also here consider using helpers in `github.com/elastic/beats/dev-tools/mage`.

Reply:
The closest helper I could find is `CreateDir`, but it expects a full file path.